graphium/utils/tensor.py

```diff
@@ -251,7 +251,7 @@ def nan_std(input: Tensor, unbiased: bool = True, **kwargs) -> Tensor:
         output: The resulting standard deviation of the tensor
     """
 
-    return torch.sqrt(nan_var(input=input, unbiased=unbiased, **kwargs))
+    return torch.sqrt(nan_var(input=input, unbiased=unbiased, **kwargs).float())
```

Comment:
`torch.sqrt` not implemented for `float16`

Reply:
Same as above. Re-casting to the original dtype.